sqlalchemy_continuum/manager.py

```diff
@@ -134,6 +136,9 @@ def reset(self):
             'after_flush': self.after_flush,
             'after_commit': self.clear,
             'after_rollback': self.clear,
+            # The below are only used by native versioning
+            'after_begin': self.after_begin,
```

Comment:
So, The original maintainer of this repo made the concept of native versioning.  He wrote a bunch of sql triggers that fire on update/delete/insert into tables that are versioned.  He _also_ created a trigger on the transaction table, which, on insert to that table, inserts into a temporary_transaction table within the database tx.  The reason for this is mostly because with a transactional context in postgres (depending on what isolation level your tx's have) you could query a table and get the mutations that occurred in other pending tx's.

So, the workflow in his mind, and what happens in continuum, was something like this:
1) session.begin()  [continuum does nothing]
2) session.add(User(...))
3) session.flush() [continuum creates a temp table called temporary_transaction, and inserts a row into audit_transaction with all the user_id, origin_app, and remote_addr columns filled, before the ORM flushes and does its insert]
4) session.commit() [postgres cleans up the temporary table]

The above works in the case where you are using ORM features only, which was his use-case afaik.

However, our goal is to enable bulk operations that are not supported by the ORM but exist in sqlalchemy.  This includes things like:
```
Query.update(...) # update users set name = 'something' where true;
session.execute(insert(...)) # insert into table_name(....) values (....);
Query.bulk_update_mappings(...) # Another way to do bulk updates
```
etc.

Also, it would be nice to just allow any query to trigger versioning properly, since native_versioning makes postgres the owner of versioning.  So these types of queries should be supported too:
```
session.execute("update users set name = 'x';")
session.execute("insert into users(name) values ('x', 'y', 'z');")

-- tunnel into postico
begin;
insert into audit_transactions(user_id) values (my_id);
insert into users(name) values ('x', 'y', 'z');
commit;
--break the tunnel
```

The solution I have here makes ALL the above, including the nice-to-haves, work.  I do this by just inserting an audit_transactions row when the transaction opens, and filling out the ancillary values (user_id, origin_app, etc) just before committing.

All the above working is a big upside - also a big upside is that every db transaction will actually have an audit_transactions row, which is more consistent (before it was only db transactions that had ORM session changes)

The only downside is that there will be no way via the _session_ to access the audit data written within a transaction.  TBH you shouldn't be doing that anyway, but if you really had to you could write a SQL query.